### PR TITLE
Makes accessing applications directly possible

### PR DESCRIPTION
**Pull request by Paalar**
- Allows for accessing directly by url
- Allows for accessing by redirect (should, but not sure how to test)
- Displays page 404 when nothing is retrieved
- Displays spinner while fetching

### CODE REVIEW

File: frontend/src/components/AdminApplicationOverview/reducer.ts [hunk 1]
```diff
@@ -37,13 +41,13 @@ export const ApplicationReducer = (
...
       };
-    case 'SET_APPLICATION_DATA': {
+    case SET_APPLICATION_DATA: {
       return {
         ...state,
         registeredApplication: action.payload,
       };
     }
-    case 'REMOVE_APPLICATION_DATA': {
+    case REMOVE_APPLICATION_DATA: {
```
Comment (FluidSense) [inline frontend/src/components/AdminApplicationOverview/reducer.ts]: This does technically not belong in this PR.

File: backend/controllers/applicationController.py [hunk 2]
```diff
@@ -21,14 +21,14 @@ def getApplicationBySelf():
...
-    return jsonify(userApplication.to_json()) if userApplication else Response(json.dumps({}), 200)
+    return jsonify(userApplication.to_json()) if userApplication else Response(json.dumps({}), 404)
 
 
 @application.route("/byUser/<userid>")
 @requiresAdmin
 def getApplicationByUser(userid):
     userApplication = applicationService.getApplicationByUserId(userid)
-    return jsonify(userApplication.to_json()) if userApplication else Response(json.dumps({}), 200)
+    return jsonify(userApplication.to_json()) if userApplication else Response(json.dumps({}), 404)
```
Comment (FluidSense) [inline backend/controllers/applicationController.py]: This is wrong, 404 indicates url does not exist

File: frontend/src/components/AdminApplication/reducer.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    case FETCH_APPLICATION_DIRECTLY_FAILED: {
+      return {
+        ...state,
+        api: {
+          status: 404,
+        },
+        application: initialState.application,
+      };
+    }
+    case RESET_REVIEW_APPLICATION_STATUS_CODE: {
```
Comment (FluidSense) [inline frontend/src/components/AdminApplication/reducer.ts]: Is this ever triggered?

File: frontend/src/components/AdminApplicationOverview/reducer.ts (hunk 1 above)
Comment (Paalar) [reply frontend/src/components/AdminApplicationOverview/reducer.ts]: Technically not, but it shouldn't have been like that from the beginning

File: frontend/src/components/AdminApplication/reducer.ts (hunk 3 above)
Comment (Paalar) [reply frontend/src/components/AdminApplication/reducer.ts]: Good catch, forgot to use this hehe
